scripts/generate_config.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python3
+# type: ignore
+
+"""Generate config file from local changes or command line.
+
+    MBEDTLS_USER_CONFIG_FILE can user-provided configuration file. This script can save local
+    changes of `include/mbedtls/mbedtls_config.h` as user-provided configuration file. Or generate
+    user-provided configuration file according to command line input.
+
+    To use it, export `CFLAGS="-I$PWD -DMBEDTLS_USER_CONFIG_FILE=\"<{headerfile}>\""
+
+"""
+
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0
+##
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+##
+# http://www.apache.org/licenses/LICENSE-2.0
+##
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import re
+import argparse
+from subprocess import check_output
+from textwrap import dedent
+
+
+from unidiff import PatchSet
+from config import ConfigFile
+
+def get_changed(file_path):
+    """Get changed config in config.h"""
+    patch = PatchSet.from_string(
+                check_output(['git', 'diff', '--', file_path],
+                             universal_newlines=True))
+
+    def iter_lines():
+        for i in patch[0]:
+            yield from i.target_lines()
+    _define_line_regexp = (r'(?P<indentation>\s*)' +
+                           r'(?P<commented_out>(//\s*)?)' +
+                           r'(?P<define>#\s*define\s+)' +
+                           r'(?P<name>\w+)' +
+                           r'(?P<arguments>(?:\((?:\w|\s|,)*\))?)' +
+                           r'(?P<separator>\s*)' +
+                           r'(?P<value>.*)')
+    regex = re.compile(_define_line_regexp)
+    for line in iter_lines():
+        if line.is_added or line.is_removed:
+            line = str(line)[1:].strip()
+            m = regex.match(line)
+            yield not m['commented_out'], m['name'], m['value']
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    enable_arg_pair_regexp = re.compile(r'(?P<name>\w+)=?(?P<value>.*)')
+    disable_arg_pair_regexp = re.compile(r'(?P<name>\w+)')
+
+    def enable_arg(arg):
+        m = enable_arg_pair_regexp.match(arg)
+        assert m, arg
+        return True, m['name'], m['value']
+
+    def disable_arg(arg):
+        m = disable_arg_pair_regexp.match(arg)
+        assert m, arg
+        return False, m['name'], ''
```

Comment:
```suggestion
        return False, m['name'], ''

```
Add a new line for better separation.